### PR TITLE
Add 'public' option to AppManager upload command

### PR DESCRIPTION
**Pull request by ADjenkov**
AppManager now supports public app distribution. 

Public option is suitable if your application already has built-in authentication mechanism or if you want to grant anonymous access to your app.AppManager user credentials will not be required to download the app.

Use private option (public not set) if you want AppManager to restrict the access to your application. AppManager user credentials will be required to download the app (the usual flow, only users in the groups can download the app)

### CODE REVIEW

**Comment by ADjenkov** [comment]
@rosen-vladimirov We've talk about it earlier ,but can you confirm it is ok that your changes about INativescriptServiceContract/NativescriptService will be included in this PR?

@ikoevska could you take a look at the .md files changes, thanks in advance. 

PS:I will  include later a warning message if --publish is not set but --public is set.

**Comment by rosen-vladimirov** [comment]
@ADjenkov, please revert the changes in server-api regarding NativeScript migration. We need these calls :) Just add your IsPublic to PublishSettings and keep the rest of the server-api

**Comment by teobugslayer** [comment]
Server.INativescriptServiceContract is removed. I guess you have generated the server APIs against server from the wrong branch. Can you regenerate them from TEST?

File: lib/server-api.d.ts [hunk 1]
```diff
@@ -256,9 +256,6 @@ declare module Server{
 		importProvision(provision: any): IFuture<Server.ProvisionData>;
 		removeProvision(identifier: string): IFuture<void>;
 	}
-	interface INativescriptServiceContract{
```
Comment (Fatme) [inline lib/server-api.d.ts]: Do you need to remove this? It seems that the swagger api is generated based on branch from Ice repo that is not rebased on release.

**Comment by Mitko-Kerezov** [comment]
:+1:

**Comment by teobugslayer** [comment]
:+1:

File: docs/man_pages/publishing/appmanager-upload-android.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ WARNING: This command is not applicable to mobile website projects. To view the
 * `--certificate` - Sets the certificate that you want to use for code signing your Android app. You can set a certificate by index or name. <% if(isHtml) { %>To list available certificates, run `$ appbuilder certificate`<% } %> 
 * `--download` - If set, downloads the application package to the root of the project.
 * `--publish` - If set, the application package will be automatically published and visible for the members of its distribution groups. To add distribution groups, use `--groups` option.
+* `--public` - If set, the published application will be available for download without requiring AppManager user credentials.
```
Comment (ikoevska) [inline docs/man_pages/publishing/appmanager-upload-android.md]: If set, the published application will be publicly available to download and install via URL.

Comment (ikoevska) [inline docs/man_pages/publishing/appmanager-upload-android.md]: Warning for when --publish is not specified:

You have not set the --publish switch. Your app will become publicly available after you publish it manually in Telerik AppManager.

Comment (ikoevska) [inline docs/man_pages/publishing/appmanager-upload-android.md]: Message for when the upload is complete, the app is **published** and we get the install URI:

Your app has been published successfully and is now available to download and install from: \<URL\>

**Comment by ikoevska** [comment]
Btw, I have noticed that this is not the help content I have provided in this PR - https://github.com/Icenium/icenium-cli/pull/1033#issuecomment-103860899 The wording is very different and the HTML-specific extended help is missing.

**Comment by ADjenkov** [comment]
The changes you have provided  in this PR - #1033 are not even on master branch. That means that my PR was not merged correctly or my commits were not actually commited...I'm confused

**Pull request merged**